docs/topics/tour/kotlin-tour-intermediate-functions.md

```diff
@@ -0,0 +1,935 @@
+}
+
+fun main() {
+    val product = Product()
+    val priceInEuros = product.getPriceInEuros()
+
+    if (priceInEuros != null) {
+        println("Price in Euros: €$priceInEuros")
+        // Price in Euros: €85.0
+    } else {
+        println("Price information is not available.")
+    }
+}
+```
+{initial-collapse-state="collapsed" collapsible="true" collapsed-title="Example solution" id="kotlin-tour-scope-functions-solution-1"}
+
+### Exercise 2 {initial-collapse-state="collapsed" collapsible="true" id="scope-functions-exercise-2"}
+
+You have an `updateEmail` function that updates the email address of one of your users. Use the `.apply()` scope function
+to update the email address and the `.also()` scope function to print a log message: `Updating email for user with ID: ${it.id}`.
+
+|---|---|
+```kotlin
+data class User(val id: Int, var email: String)
+
+fun updateEmail(user: User, newEmail: String): User = // Write your code here
+
+fun main() {
+    val user = User(1, "old_email@example.com")
+    val updatedUser = updateEmail(user, "new_email@example.com")
+    // Updating email for user with ID: 1
+    
+    println("Updated User: $updatedUser")
+    // Updated User: User(id=1, email=new_email@example.com)
+}
+
+```
+{validate="false" kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="kotlin-tour-scope-functions-exercise-2"}
+
+|---|---|
+```kotlin
+data class User(val id: Int, var email: String)
+
+fun updateEmail(user: User, newEmail: String): User = user.apply {
+    this.email = newEmail
+}.also { println("Updating email for user with ID: ${it.id}") }
+
+fun main() {
+    val user = User(1, "old_email@example.com")
+    val updatedUser = updateEmail(user, "new_email@example.com")
+    // Updating email for user with ID: 1
+
+    println("Updated User: $updatedUser")
+    // Updated User: User(id=1, email=new_email@example.com)
+}
+```
+{initial-collapse-state="collapsed" collapsible="true" collapsed-title="Example solution" id="kotlin-tour-scope-functions-solution-2"}
+
+## Lambda expressions with receiver
+
+In the beginner's tour, you learned how to use [lambda expressions](kotlin-tour-functions.md#lambda-expressions). Lambda expressions can also have a receiver, 
+which means that the lambda expression can access any member functions or properties of the receiver object without having
+to explicitly specify it each time. Without these additional references, your code is easier to read and maintain.
+
+The syntax for a lambda expression with receiver is different when you define the function type: write the receiver object
+that you want to extend followed by a `.` and the rest of your function type definition. For example: `MutableList<Int>.() -> Unit`.
+
+This function type has:
+
+* `MutableList<Int>` as the receiver type.
+* No function parameters within the parentheses `()`.
+* No return value: `Unit`.
+
+> Lambda expressions with receiver are also called function literals with receiver.
```

Comment:
This hint could be right after the 1st paragraph, I think here it's too late to add extra terms, in the middle of actual concept explanation